Update ReadMe & Files to support TCX data from various sources

mjunkmyjunk opened the pull request:
Update README to show how decode works
Update some files to docode TCX files from various sources. Now working with Strava and Tapiriik.sync

mjunkmyjunk committed:
Merge pull request #1 from mjunkmyjunk/mjunkmyjunk-patch-1

Update README.md

mjunkmyjunk committed:
Merge branch 'FitnessKit:master' into master

Commit:
Update README to show how to use decode function and within a do-catch to have error messages printed out
Modified TcxFile to allow for alernate DateTime from different sources
Modified TrainingCenterDatabase.swift to allow for optional String to accomodate different TCX data sources

Commit:
fixed typo

khoogheem commented on Sources/TcxDataProtocol/TcxFile.swift:
```diff
@@ -79,11 +87,29 @@ extension TcxFile {
+            let dateStr = try container.decode(String.self)
+            let len = dateStr.count
+            var date: Date? = nil
+            if len == 24 { // "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"
```
put vertical spacing between the params and statements

khoogheem commented on Sources/TcxDataProtocol/TcxFile.swift:
```diff
@@ -79,11 +87,29 @@ extension TcxFile {
+            } else {       // "yyyy-MM-dd'T'HH:mm:ss'Z'"
+                date = formatterFullTime.date(from: dateStr)
+            }
+            guard let date_ = date else {
```
you don't really need too check the Len here.. you can

```
if let date = formatterFractionalSeconds.date(from: dateStr) {
  return date
}

if let date = formatterFullTime.date(from: dateStr) {
 return date 
}

if none of the formatters work
throw dataCorupt

khoogheem commented on README.md:
```diff
@@ -32,8 +32,40 @@ let tcxUrl = URL(fileURLWithPath: "TestFile" + ".tcx")
+  
+  for activity in 0..<numActivities {    
+    guard let numLaps = tcxFile.database.activities?.activities?[activity].lap.count else {  }
+    for laps in 0..<numLaps {
```
vert space between the guard and the loop..

also I would switch to just doing
```
-- If you really want to check Empty but that is not needed if you change to the for in loop
guard let activities = tcxFile.database.activities?.activities, !activities.isEmpty else { return }

for activity in activities {

}
```

mjunkmyjunk committed:
Update TcxFile.swift

Remove Len check.

mjunkmyjunk committed:
Handle Different Time Formats

Some sites may have different time formats in the TCX files. 
This enables 2 diff time formats.

"yyyy-MM-dd'T'HH:mm:ss.SSS'Z'" and
"yyyy-MM-dd'T'HH:mm:ss'Z'"

khoogheem commented:
I see you marked this now as draft... does the new PR cover all of this? 
and do you just want to close it?

This was not merged due to no response to suggestions regarding the readme..

mjunkmyjunk commented:
Yes. the new PR covers all the same changes Except for the Readme which I do not know how to resolve using your suggestions

The new PR removes the Readme from consideration.

khoogheem commented:
> Yes. the new PR covers all the same changes Except for the Readme which I do not know how to resolve using your suggestions
> 
> The new PR removes the Readme from consideration.

the big part was you can change:
```
guard let numActivities = tcxFile.database.activities?.activities?.count else { return  }
   
   for activity in 0..<numActivities {    
   }
```

to something like:

```
guard let  activities = tcxFile.database.activities?.activities else { return  }
   
   for activity in activities {    
   }
```

no reason to do count

mjunkmyjunk commented:
I will give this a try and if it works, I will re-do the Readme w/ your suggested changes.
TQ again

khoogheem commented:
1.0.2 Pod and SPM should include changes
